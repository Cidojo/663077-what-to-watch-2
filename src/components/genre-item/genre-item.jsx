import * as React from 'react';
import * as PropTypes from 'prop-types';

const GenreItem = (props) => {
  const {
    currentGenre,
    genre,
    onGenreTabClick
  } = props;

  const handleGenreTabClick = (e) => {
    e.preventDefault();

    onGenreTabClick(e.currentTarget.textContent);
  };

  return (
    <li
      className={`catalog__genres-item${currentGenre === genre ? ` catalog__genres-item--active` : ``}`}
    >
      <a
        href="#"
        className="catalog__genres-link"
        onClick={handleGenreTabClick}
      >
        {genre}
      </a>
    </li>
  );
};

GenreItem.propTypes = {
  currentGenre: PropTypes.string,
  genre: PropTypes.string,
  onGenreTabClick: PropTypes.func.isRequired,
};

export {GenreItem};
