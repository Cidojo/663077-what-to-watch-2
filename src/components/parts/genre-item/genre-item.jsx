import * as React from 'react';
import * as PropTypes from 'prop-types';

const GenreItem = (props) => {
  const {
    isActive,
    genre,
    onGenreTabClick
  } = props;

  const handleGenreTabClick = (e) => {
    e.preventDefault();

    onGenreTabClick(e.currentTarget.textContent.trim());
  };

  return (
    <li className={`catalog__genres-item${isActive ? ` catalog__genres-item--active` : ``}`} >
      <a
        href="#"
        className="catalog__genres-link"
        onClick={handleGenreTabClick}
      >
        {genre}
      </a>
    </li>
  );
};

GenreItem.propTypes = {
  isActive: PropTypes.bool,
  genre: PropTypes.string,
  onGenreTabClick: PropTypes.func.isRequired,
};

GenreItem.defaultProps = {
  isActive: false,
  genre: ``,
  onGenreTabClick: () => {}
};

export {GenreItem};
