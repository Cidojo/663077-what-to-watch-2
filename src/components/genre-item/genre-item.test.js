import * as React from 'react';
import {GenreItem} from './genre-item.jsx';
import * as renderer from 'react-test-renderer';

it(`should render component without errors`, () => {
  const tree = renderer
    .create(<GenreItem
      currentGenre={``}
      genre={``}
      onGenreTabClick={() => {}}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
