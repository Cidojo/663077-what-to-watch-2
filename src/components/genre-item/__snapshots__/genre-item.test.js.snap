// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component without errors 1`] = `
<li
  className="catalog__genres-item"
>
  <a
    className="catalog__genres-link"
    href="#"
    onClick={[Function]}
  >
    
  </a>
</li>
`;
